app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -142,6 +142,12 @@ export function emailValid(email) {
   return re.test(email);
 }
 
+export function hostnameValid(hostname) {
+  // see:  https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+  const re = /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/;
```

Comment:
I should note we have had trouble with regular expressions for hostnames in the past. This might work well enough for us, but it's something we should keep in mind.